content/en/docs/ops/common-problems/network-issues/index.md

```diff
@@ -233,6 +233,115 @@ server {
 }
 {{< /text >}}
 
+## 404 or 503 error while accessing headless services
+
+Assume Istio is installed with the following configuration:
+
+- `mTLS mode` set to `STRICT` within the mesh
+- `meshConfig.outboundTrafficPolicy.mode` set to `ALLOW_ANY`
+
+Consider `nginx` is deployed as a `StatefulSet` in the default namespace and a corresponding `Headless Service` is defined as shown below:
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx
+  labels:
+    app: nginx
+spec:
+  ports:
+  - port: 80
+    name: web      # Port name
+  clusterIP: None  # Creates a Headless Service
+  selector:
+    app: nginx
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  serviceName: "nginx"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: k8s.gcr.io/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+{{< /text >}}
+
+Let us assume we have a [sleep]({{< github_tree >}}/samples/sleep) pod `Deployment` as well in the default namespace.
+
+Now, let us consider 3 possible scenarios based on the port `name` defined in the headless service:
+
+1. Port `name` set to `web` (as shown in the example above) or any other custom name
+
+    - Request without Host header:
+
+        When `nginx` is accessed without the Host header, the sidecar proxy on the client-side fails to find the route entry to `nginx` and fails with `HTTP 404 NR`.
+
+        {{< text bash >}}
```

Comment:
These scripts should be rewritten so they can be cut and pasted into a terminal window. At least sort of. I know you would need to set/find the pod name and IP.  Can you just run the `curl` instead of the `sh`? You will have to provide two for the last bash box, one for each curl.

Reply:
In other guides, we set 
```export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath='{.items..metadata.name}')```
and then
`kubectl exec -it $SOURCE_POD -c sleep -- sh` or curl directly

Reply:
Ok, right. I can make these changes. So, I will need to add the SOURCE_POD and also something similar for the pod IP of the nginx sts.